x-pack/plugins/spaces/server/lib/spaces_url_parser.js

```diff
@@ -3,33 +3,38 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { DEFAULT_SPACE_ID } from '../../common/constants';
 
-export function getSpaceUrlContext(requestBasePath = '/', serverBasePath = '/') {
+export function getSpaceIdFromPath(requestBasePath = '/', serverBasePath = '/') {
   let pathToCheck = requestBasePath;
 
   if (serverBasePath && serverBasePath !== '/' && requestBasePath.startsWith(serverBasePath)) {
     pathToCheck = requestBasePath.substr(serverBasePath.length);
   }
   // Look for `/s/space-url-context` in the base path
-  const matchResult = pathToCheck.match(/^\/s\/([a-z0-9\-]+)/);
+  const matchResult = pathToCheck.match(/^\/s\/([a-z0-9_\-]+)/);
```

Comment:
Did we mean to add an `_` here? I'm not seeing an `_` being allowed in the spaceSchema `id`.

Reply:
ah yep, that was a partial change that I didn't finish. thanks!